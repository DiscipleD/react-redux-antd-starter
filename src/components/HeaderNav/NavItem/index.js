/**
 * Created by jack on 16-9-23.
 */

import React from 'react';
import { Link } from 'react-router';

class NavItem extends React.Component {
	render() {
		const item = this.props.item;
		const navItemElement = item.link
			? <Link to={item.link}>{item.label}</Link>
				: typeof item.fn === 'function'
				? <a onClick={item.fn}>{item.label}</a>
				: item.label;
		return (
			<li className="nav-item">{navItemElement}</li>
		)
	}
}

export default NavItem;
