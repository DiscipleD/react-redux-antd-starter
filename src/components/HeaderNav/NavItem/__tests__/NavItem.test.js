/**
 * Created by jack on 16-9-25.
 */

import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-addons-test-utils';

import NavItem from '../index';

describe('NavItem Component', () => {
	it('Plain Item', () => {
		const itemSetting = {
			name: 'a',
			label: 'aa'
		};
		const navItem = <NavItem item={itemSetting}/>;
		const navItemElement = ReactTestUtils.renderIntoDocument(navItem);
		const navItemNode = ReactDOM.findDOMNode(navItemElement);

		expect(navItemNode.textContent).toBe(itemSetting.label);
		expect(renderer.create(navItem).toJSON()).toMatchSnapshot();
	});

	it('Link Item', () => {
		const itemSetting = {
			name: 'a',
			label: 'aa',
			link: '/a'
		};
		const navItem = <NavItem item={itemSetting}/>;
		const navItemElement = ReactTestUtils.renderIntoDocument(navItem);
		const navItemNode = ReactDOM.findDOMNode(navItemElement);

		// navItemNode link doesn't include href attribute
		expect(navItemNode.querySelector('a')).toBeDefined();
		expect(renderer.create(navItem).toJSON()).toMatchSnapshot();
	});

	it('Function Item', () => {
		const itemSetting = {
			name: 'a',
			label: 'aa',
			fn: jest.fn()
		};
		const navItem = <NavItem item={itemSetting}/>;
		const navItemElement = ReactTestUtils.renderIntoDocument(navItem);
		const navItemNode = ReactDOM.findDOMNode(navItemElement);

		ReactTestUtils.Simulate.click(navItemNode.querySelector('a'));
		expect(itemSetting.fn).toBeCalled();
		expect(renderer.create(navItem).toJSON()).toMatchSnapshot();
	});
});

