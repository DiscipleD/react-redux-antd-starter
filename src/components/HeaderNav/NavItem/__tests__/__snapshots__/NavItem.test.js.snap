exports[`NavItem Component Function Item 1`] = `
<li
  className="nav-item">
  <a
    onClick={[Function mockConstructor]}>
    aa
  </a>
</li>
`;

exports[`NavItem Component Link Item 1`] = `
<li
  className="nav-item">
  <a
    onClick={[Function bound handleClick]}
    style={Object {}}>
    aa
  </a>
</li>
`;

exports[`NavItem Component Plain Item 1`] = `
<li
  className="nav-item">
  aa
</li>
`;
